pulpcore/cli/python/content.py

```diff
@@ -0,0 +1,159 @@
+    click.option(
+        "--description", help="A longer description of the package that can run several paragraphs."
+    ),
+    click.option(
+        "--keywords",
+        help="Additional keywords to be used to assist searching for the package in a larger"
+        " catalog.",
+    ),
+    click.option("--home_page", help="The URL for the package's home page."),
+    click.option(
+        "--download_url",
+        help="Legacy field denoting the URL from which this package can be downloaded.",
+    ),
+    click.option(
+        "--author",
+        help="Text containing the author's name. Contact information can also be added, separated"
+        " with newlines.",
+    ),
+    click.option("--author_email", help="The author's e-mail address."),
+    click.option(
+        "--maintainer",
+        help="The maintainer's name at a minimum; additional contact information may be provided.",
+    ),
+    click.option("--maintainer_email", help="The maintainer's e-mail address."),
+    click.option("--license", help="Text indicating the license covering the distribution"),
+    click.option(
+        "--requires_python",
+        help="The Python version(s) that the distribution is guaranteed to be compatible with.",
+    ),
+    click.option(
+        "--project_url",
+        help="A browsable URL for the project and a label for it, separated by a comma.",
+    ),
+    click.option(
+        "--platform",
+        help="A comma-separated list of platform specifications, "
+        "summarizing the operating systems supported by the package.",
+    ),
+    click.option(
+        "--supported_platform",
+        help="Field to specify the OS and CPU for which the binary package was compiled.",
+    ),
+    click.option(
+        "--requires_dist",
+        help="A JSON list containing names of some other distutils project required by this"
+        " distribution.",
+    ),
+    click.option(
+        "--provides_dist",
+        help="A JSON list containing names of a Distutils project which is contained within this"
+        " distribution.",
+    ),
+    click.option(
+        "--obsoletes_dist",
+        help="A JSON list containing names of a distutils project's distribution which this "
+        "distribution renders obsolete, meaning that the two projects should not be "
+        "installed at the same time.",
+    ),
+    click.option(
+        "--requires_external",
+        help="A JSON list containing some dependency in the system that the distribution is to be"
+        " used.",
+    ),
+    click.option(
+        "--classifiers", help="A JSON list containing classification values for a Python package."
+    ),
+]
+content.add_command(
+    list_command(
+        decorators=[
+            click.option("--filename", type=str),
+        ]
+    )
+)
+content.add_command(show_command(decorators=[href_option]))
+content.add_command(create_command(decorators=create_options + python_content_options))
```

Comment:
I know, this is probably, how the python plugin works. But I would expect that is tell it to "take that artifact and turn it into a python package" and the server would analyse the file to get all the information from `setup.py`.

Reply:
You are right, the upload api does get the information from the package. It even overwrites whatever you pass in with the info from the package, so having these options is misleading, I'll remove them.